ui/src/app/cohort-search/list-overview/list-overview.component.tsx

```diff
@@ -0,0 +1,426 @@
+    borderRadius: '3px',
+    float: 'right',
+  },
+  totalCount: {
+    marginTop: 0,
+    fontSize: '16px',
+    fontWeight: 'bold'
+  },
+  totalError: {
+    background: '#f7981c',
+    color: '#ffffff',
+    padding: '0.25rem 0.5rem',
+    borderRadius: '5px',
+    marginBottom: '0.5rem'
+  },
+  actionIcon: {
+    float: 'right',
+    margin: '0 1rem 0 0',
+    minWidth: 0,
+    padding: 0,
+    color: '#262262',
+    cursor: 'pointer'
+  },
+  disabled: {
+    opacity: 0.4,
+    cursor: 'not-allowed',
+    pointerEvents: 'none'
+  },
+  cardContainer: {
+    boxShadow: '0 0.125rem 0 0 #d7d7d7',
+    border: '1px solid #d7d7d7',
+    borderBottom: 'none',
+    borderTop: 'none',
+    borderRadius: '3px',
+  },
+  card: {
+    background: '#ffffff',
+    marginBottom: 0,
+    marginTop: 0,
+  },
+  cardHeader: {
+    color: '#302C71',
+    fontSize: '13px',
+    borderBottom: 'none',
+    padding: '0.5rem 0.75rem',
+  },
+  error: {
+    background: '#f7981c',
+    color: '#ffffff',
+    fontSize: '11px',
+    border: '1px solid #ebafa6',
+    borderRadius: '3px',
+    marginBottom: '0.25rem',
+    padding: '3px 5px'
+  },
+  invalid: {
+    background: '#f5dbd9',
+    color: '#565656',
+    fontSize: '11px',
+    border: '1px solid #ebafa6',
+    borderRadius: '3px',
+    marginBottom: '0.25rem',
+    padding: '3px 5px'
+  },
+});
+
+interface Props {
+  searchRequest: any;
+  update: number;
+}
+
+interface State {
+  error: boolean;
+  loading: boolean;
+  total: number;
+  chartData: any;
+  saveModal: boolean;
+  name: string;
+  description: string;
+  nameTouched: boolean;
+  saving: boolean;
+  deleting: boolean;
+  stackChart: boolean;
+  showConflictError: boolean;
+  saveError: boolean;
+  cohort: Cohort;
+  apiCallCheck: number;
+}
+
+export const ListOverview = withCurrentWorkspace()(
+  class extends React.Component<Props, State> {
+    dropdown: any;
+    constructor(props: Props) {
+      super(props);
+      this.state = {
+        error: false,
+        loading: false,
+        total: undefined,
+        chartData: undefined,
+        saveModal: false,
+        name: undefined,
+        description: undefined,
+        nameTouched: false,
+        saving: false,
+        deleting: false,
+        stackChart: false,
+        showConflictError: false,
+        saveError: false,
+        cohort: undefined,
+        apiCallCheck: 0,
+      };
+    }
+
+    componentDidMount(): void {
+      if (currentCohortStore.getValue()) {
+        this.setState({cohort: currentCohortStore.getValue()});
+      }
+      this.getTotalCount();
+    }
+
+    componentDidUpdate(prevProps: Readonly<Props>): void {
+      if (this.props.update > prevProps.update && !this.hasErrors) {
+        this.setState({loading: true, error: false});
+        this.getTotalCount();
+      }
+    }
+
+    getTotalCount() {
+      try {
+        const {searchRequest} = this.props;
+        const localCheck = this.state.apiCallCheck + 1;
```

Comment:
Yeah, this is an interesting one, not sure if there's a better way to do it. I'm using this variable to check if more api calls are made before this call returns. If `apiCallCheck` has changed in state since `localCheck` was set, another call has been made and I don't update state.

The problem occurs when you make several changes quickly that trigger a count calculation (toggle the temporal switch or suppress/enable search items). The first call would return and set loading to false and show an outdated count while to other calls are still pending. Eventually, the right count shows up, but it was confusing when the spinner goes away then the count changes again (possibly multiple times) several seconds later.

This is my attempt of making sure the state only gets updated with the most recent api call. Seems to work well, but there may be a better way to do it.